common/views/components/GlobalAlertContext/GlobalAlertContext.tsx

```diff
@@ -0,0 +1,49 @@
+import { Component, createContext, ReactNode } from 'react';
+import { HTMLString } from '@weco/common/services/prismic/types';
+const GlobalAlertContext = createContext(null);
+
+type GlobalObject = {
+  isShown: boolean | null;
+  routeRegex: string | null;
+  text: HTMLString;
+};
+
+type Props = {
+  value: GlobalObject;
+  children: ReactNode;
+};
+
+type State = {
+  isShown: boolean | null;
+  routeRegex: string | null;
+  infoBanner: boolean;
+};
+
+export class GlobalAlertContextProvider extends Component<Props, State> {
+  setInfoBanner = (value: boolean) => {
+    this.setState({
+      ...this.state,
+      infoBanner: value,
+    });
+  };
+
+  // initial context value
+  state = {
+    infoBanner: false,
+    isShown: this.props.value.isShown,
+    routeRegex: this.props.value.routeRegex,
+    text: this.props.value.text,
+    setInfoBanner: this.setInfoBanner,
```

Comment:
children components can set state which can be passed around to update state  to other components.

Reply:
Should we convert this to hooks and a functional component while we're at it?

Reply:
Do you mean converting this existing file to a functional component ? and using hooks to save set of setInfoBanner?